buzzmobile/tests/test_utils/launch_tools.py

```diff
@@ -44,13 +51,13 @@ def launcher(func):
         def new_test(self):
             """Wrapper around the user provided test that runs a launch file.
             """
-            launch = ROSLauncher(full_name)
+            os.environ['ROS_MASTER_URI'] = self.rosmaster_uri
+            launch = ROSLauncher(full_name, port=self.port)
```

Comment:
If I understand correctly, this means that while both of the examples you wrote are allowed, the following would not be:

```python
@mock_node(*node_args)
@mock_node(*other_args)
@mock_node(*more_of_these)
@with_launch_file(*args)
def test func():
    ...
```

or 

```python
@mock_node(*node_args)
@mock_node(*other_args)
@with_launch_file(*args)
@mock_node(*more_of_these)
def test func():
    ...
```

And yeah can you make a comment for that?